### ISSUE TITLE
Tackle confusion on multiple Instance types

### ISSUE THREAD

**Issue by pdesoyres-cc** [CleverCloud/clever-components#917]
When working on the new logs view, I discovered that some other components already use `Application` and `Instance` concepts.
The `cc-logs-instances` component used inside the new logs view displays a list of Instances.
Those instances come from the V4 API that doesn't have the same shape as the one already designed for other components.

This creates confusion between those types.

## The actual situation

3 types are involved:

```ts
interface Application {
  name: string;
  link: string;
  instance: Instance;
  zone: Zone;
}

interface Instance {
  flavourName: string;
  count: number;
}

interface InstancesState {
  running: Instance[];
  deploying: Instance[];
}
```

All those three types are located in `common.types.d.ts`.

2 components are involved:

* `cc-addon-linked-apps`
  * uses `Application`
  * uses the field `variant` of the application instance even though it is not defined on the `Instance` interface.
  * It uses none of the fields defined in the `Instance` interface.
  * This component is fed by its associated smart component
  * It takes its data from `/v2/organisations/${owner_id}/addons/${addon_id}/applications` API.
* `cc-tile-instances`
  * uses `InstancesState`
  * It is used directly in the console (no smart component)
  * Data are taken from `/v2/organisations/${owner_id}/instances` API
  * Data from this API is transformed to get
    * the instance count (`count` property)
    * the `flavorName` is fed with `flavor.name`

To sum up, we have  
* `cc-addon-linked-apps` component which doesn't use any of the fields defined in the `Instance` interface.
   Instead, it uses an undefined field.
* `cc-tile-instances` component which uses something converted from the instances API.

### `cc-addon-linked-apps`

* create its own `types.d.ts` file
* create a `LinkedApplication` interface

```ts
interface LinkedApplication {
  name: string;
  link: string;
  variantName: string;
  variantLogoUrl: string;
  zone: Zone;
}
```

### `cc-tile-instances`

* move types into its own `types.d.ts` file
* rename `Instance` type to `InstanceState`

**Comment by pdesoyres-cc** [comment]
see #928 & #919
